arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptyList.kt

```diff
@@ -383,6 +383,12 @@ public class NonEmptyList<out A>(
       map(head, b.head, c.head, d.head, e.head, f.head, g.head, h.head, i.head, j.head),
       tail.zip(b.tail, c.tail, d.tail, e.tail, f.tail, g.tail, h.tail, i.tail, j.tail, map)
     )
+
+  public inline fun <B : Comparable<B>> minBy(selector: (A) -> B): A =
+    minByOrNull(selector) ?: head
```

Comment:
Dont need `?: head`, just `minByOrNull(selector)!!`

Reply:
`!!` should indeed be safe here, I think it results in a faster binary but not 100% sure actually.